analysis/builders.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/awslabs/ar-go-tools/analysis/config"
+	"github.com/awslabs/ar-go-tools/analysis/dataflow"
+	"github.com/awslabs/ar-go-tools/analysis/loadprogram"
+	"github.com/awslabs/ar-go-tools/analysis/ptr"
+	"github.com/awslabs/ar-go-tools/internal/formatutil"
+)
+
+// BuildWholeProgramTarget loads the target specified by the list of files provided. Return an analyzer state that has
+// been initialized with the program if successful. The Target of that state will be set to the provided name.
+func BuildWholeProgramTarget(
+	c *config.State,
+	name string,
+	patterns []string,
+	options loadprogram.Options) (*loadprogram.State, error) {
+	if name != "" {
+		// If it's a named target, need to change to project root's directory to properly load the target
+		err := os.Chdir(c.Config.Root())
```

Comment:
Good point.
Since this was a change from the previous PR and I'm not sure how well it will work with larger programs with vendored dependencies we can make that change in a future PR?